1주차/131128_토/131128_김지수.java

```diff
@@ -0,0 +1,75 @@
+// Solution 1
+import java.util.*;
+class Solution1 {
+    public String solution(String X, String Y) {
+        // 각 문자열의 문자 빈도를 저장하는 Map
+        Map<Character, Integer> charFrequency = new HashMap<>();
+
+        // X 문자열의 문자 빈도만큼 증가
+        for (char x : X.toCharArray()) {
+            charFrequency.put(x, charFrequency.getOrDefault(x, 0) + 1);
+        }
+
+        // Y 문자열의 문자 빈도만큼 감소하면서 공통 문자 찾기
+        StringBuilder common = new StringBuilder();
+        for (char y : Y.toCharArray()) {
+            if (charFrequency.containsKey(y) && charFrequency.get(y) > 0) {
+                common.append(y);
+                charFrequency.put(y, charFrequency.get(y) - 1);
+            }
+        }
+
+        // 공통된 문자를 내림차순으로 정렬
+        /* 문자열은 기본타입이므로 불변! 우선 문자배열로 바꿔서 정렬 후 문자열로 바꿔주자 */
+        char[] commonChars = common.toString().toCharArray();
+        Arrays.sort(commonChars);
+        String answer = new StringBuilder(new String(commonChars)).reverse().toString();
+
+        return answer.isEmpty() ? "-1" : (answer.replaceAll("0", "").isEmpty() ? "0" : answer);
+    }
+}
+/* 시간복잡도: O(X + Y + ClogC) => C는 공통 원소 개수(만약 이게 상수면 O(X + Y)로 간략화될 수 있음) */
+
+
+// Solution 2
+class Solution2 {
+    public String solution(String X, String Y) {
+        // 결과를 저장할 StringBuilder 생성
+        StringBuilder answer = new StringBuilder();
+
+        // 숫자 0부터 9까지의 빈도를 저장할 배열 초기화
+        int[] x = {0,0,0,0,0,0,0,0,0,0};
+        int[] y = {0,0,0,0,0,0,0,0,0,0};
```

Comment:
```suggestion
        int[] x = new int[10];
        int[] y = new int[10];
```

으로 초기화하여도 int 배열의 기본값이 0이기 때문에 0으로 초기화되어서 들어간답니다 :)

Reply:
오! 그러네요~! 깜빡 잊고 있었는데 알려주셔서 감사합니다

Reply:
헉 리뷰 달러왔다가 한수 배우고 갑니다! 저도 이렇게 선언했는데